lib/document.js

```diff
@@ -555,9 +555,15 @@ function init(self, obj, doc, prefix) {
 Document.prototype.update = function update() {
   const args = utils.args(arguments);
   args.unshift({_id: this._id});
-  return this.constructor.update.apply(this.constructor, args);
+  return this.constructor.updateOne.apply(this.constructor, args);
```

Comment:
In the interest of keeping the API consistent, let's make this still call `update()` but make a separate `Document.prototype.updateOne()`. As specified in [this docs commit](https://github.com/Automattic/mongoose/commit/99813ad2c882e4bcbb4d607e75d106d2523483a1), there are a couple meaningful differences between `update()` and `updateOne()`:

1) [MongoDB version compatibility](https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/). Mongoose 5.x still [supports MongoDB 3.0.x](https://mongoosejs.com/docs/compatibility.html), but `updateOne()` requires MongoDB 3.2.
2) `update()` supports the `overwrite` option, `updateOne()` does not.

We should also add a `Document.prototype.replaceOne()` analogous to `updateOne()`.

Reply:
@vkarpov15 But in a single document, the condition will always be `_id` of the document. So it should be always `update this single document` aka `updateOne`. I think it's good to keep API consistent but adding `updateOne` may be a little bit confusing? Just my opinion though.

Re: compatibility issue: mongodb driver is actually [calling `update` internally for `updateOne`](https://github.com/mongodb/node-mongodb-native/blob/36e92f121a5c1b9c28e5cddbc8fffd4a5ed220c0/lib/operations/collection_ops.js#L1408-L1413) operation and the current node-mongodb-native driver is still [marked as supporting MongoDB 2.6](https://docs.mongodb.com/ecosystem/drivers/driver-compatibility-reference/#node-js-driver-compatibility)

Reply:
@Fonger good call, but that still leaves the issue with the `overwrite` option. Plus, refactoring is much easier when you can just replace `update()` with `updateOne()` everywhere as opposed to not replacing it when using a doc.

Reply:
I'll make the changes now. Thanks guys 👍